src/render/wiiu/SDL_rtexture_wiiu.c

```diff
@@ -180,10 +186,30 @@ int WIIU_SDL_UpdateTexture(SDL_Renderer * renderer, SDL_Texture * texture,
     /* We write the rules, and we say all textures are streaming */
     WIIU_SDL_LockTexture(renderer, texture, rect, (void**)&dst, &dst_pitch);
 
-    for (row = 0; row < rect->h; ++row) {
-        SDL_memcpy(dst, src, length);
-        src += pitch;
-        dst += dst_pitch;
+    if (length == pitch && length == dst_pitch) {
+        /* DMA requires 8 byte alignment */
+        src_aligned = ((uintptr_t)src & 7U) == 0;
+        dst_aligned = ((uintptr_t)dst & 7U) == 0;
+        /* DMA works best on textures bigger than 5KiB */
+        if ((total_size > 5120) && (src_aligned && dst_aligned)) {
+            DCFlushRange(src, total_size);
+            /* Run a single DMA transfer and wait until transfer is done */
+            while (!DMAEWaitDone(DMAECopyMem(dst, src, total_size >> 2, DMAE_SWAP_NONE)));
+        } else {
+            /* Otherwise, fallback to a single memory copy */
+            OSBlockMove(dst, src, total_size, true);
+        }
+    } else {
+        /* Flush source first */
+        DCFlushRange(src, total_size);
```

Comment:
Are these flushes necessary? IIRC `OSBlockMove` will already flush source and destination while copying.

Reply:
It's a bit faster to flush the whole src/dst at once than letting `OSBlockMove` handle it per line copy (I explicitly call `OSBlockMove` with the `flush=false` param for that reason).